deploy/helm/sumologic/conf/traces/traces.otelgateway.conf.yaml

```diff
@@ -0,0 +1,12 @@
+{{- $yamlFile := toYaml $.Values.otelgateway.config }}
+
+{{- $otelcolService := include "sumologic.metadata.name.otelcol.service" . }}
+{{- $collectorAddress := printf "%s.%s" $otelcolService .Release.Namespace }}
+{{- $yamlFile = replace "exporters.otlp.endpoint.replace" $collectorAddress $yamlFile }}
+
+{{- $otelcolHeadlessService := include "sumologic.metadata.name.otelcol.service-headless" . }}
+{{- $otelcolHeadlessServiceAddress := printf "%s.%s" $otelcolHeadlessService .Release.Namespace }}
+{{- $yamlFile = replace "exporters.loadbalancing.endpoint.replace" $otelcolHeadlessServiceAddress $yamlFile }}
+
+
+{{ tpl ($yamlFile | replace ": '{{" ": {{" | replace "}}'" "}}") . | nindent 2 }}
```

Comment:
Please consider using simplified conf (no additional logic and hacky replacements) file like this one:

https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/main/deploy/helm/sumologic/conf/logs/otelcol/config.yaml

altogether with includes in values.yaml
https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/6c94c5e81f22deb8cab97f26d0f7b18713373bce/deploy/helm/sumologic/values.yaml#L4194

Reply:
also add unit tests for the template configuration:
https://github.com/SumoLogic/sumologic-kubernetes-collection/tree/main/tests/helm/logs_otc

Reply:
I copied that code from otelagent. I assumed it's what we want to do.

Reply:
It was old approach :sweat_smile:

Reply:
> As for the replacements, I also dont't like having that if-else block. I want to redirect agent to the gateway if gateway is present or to collector if it is not present.

which `if-else` ?

Reply:
in the oletagent